adapter/slack.py

```diff
@@ -61,4 +63,19 @@ async def ask(self, request: Request):
                     raise HTTPException(status_code=400, detail=f"Slack API Error : {delete_error}")
 
             raise HTTPException(status_code=400, detail=f"Slack API Error : {e}")
+    
+    async def list_bots(self, request: Request):
+        data = await request.form()
+    
+        channel_id = data.get("channel_id")
         
+        bot_responses = self.bot_controller.fetch_chatbots()
+        
+        response_text = f"{len(bot_responses)} Active Bot(s)"
+        response_text += ''.join([f'\n- {bot_response.name} ({bot_response.id})' for bot_response in bot_responses])
+        
+        try:
+            self.app.client.chat_postMessage(channel=channel_id, text=response_text)
+            return Response(status_code=200)

```

Comment:
You can actually refactor this to return a simple JSON type response. You can compose the messages using Block Kit as noted here. This makes the test easier to compose as we're just testing Python dictionaries for the positive case.

https://api.slack.com/block-kit/building